frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx

```diff
@@ -29,6 +33,27 @@ type Props = {
 
 const EmploymentAccordion: React.FC<Props> = ({ index }: Props) => {
   const { t } = useTranslation();
+  const { getValues } = useFormContext<Application>();
+  const { applicationQuery, fetchEmployment } = useApplicationApi();
+  const [isFetchEmployeeDataEnabled, setIsFetchEmployeeDataEnabled] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    enableFetchEmployeeDataButton();
+  }, []);
+
+  const enableFetchEmployeeDataButton = () => {
+    const formDataVoucher = getValues().summer_vouchers[index];
+    setIsFetchEmployeeDataEnabled(
+      formDataVoucher.employee_name.length > 0 &&
+        formDataVoucher.summer_voucher_serial_number.length > 0
```

Comment:
Checking with mobile, so asking, can it be null?

```suggestion
formDataVoucher?.employee_name?.length && formDataVoucher?.summer_voucher_serial_number?.length
```

Reply:
No it cannot be. Button is shown only as part of the form, and form is shown only when employment voucher is added to the application in backend and returned, so we have already form data. Name is always empty string in the beginning, the same as serial number.